utils/include/MatchWildcard.hpp

```diff
@@ -25,7 +25,7 @@ template <class CHAR_T, class CHAR_COMPARE>
 			wild++;
 			str++;
 
-		} else {
+		} else if (mp && cp){
```

Comment:
If think about logic, this condition is always true. To make automatic code validator happy its better to put assert instead, cuz otherwise such check may mislead to thinking that there is bug that causes infinite loop.